versions/3.1.1.md

```diff
@@ -199,10 +199,12 @@ Note that no aspect of implicit connection resolution changes how [URIs are reso
 
 ### Data Types
 
-Data types in the OAS are based on the types supported by the [JSON Schema Specification Draft 2020-12](https://tools.ietf.org/html/draft-bhutton-json-schema-00#section-4.2.1).
-Note that `integer` as a type is also supported and is defined as a JSON number without a fraction or exponent part.
 Models are defined using the [Schema Object](#schema-object), which is a superset of JSON Schema Specification Draft 2020-12.
 
+Data types in the OAS are based on the six types supported by the [JSON Schema Specification Draft 2020-12 data model](https://tools.ietf.org/html/draft-bhutton-json-schema-00#section-4.2.1): array, boolean, null, number, object, or string.  JSON Schema keywords and `format` values operate on these six types, with certain keywords and formats only applying to a specific type.  For example, the `pattern` keyword and the `date-time` format only apply to strings, and treat any instance of the other five types as _automatically valid._  This means JSON Schema keywords and formats do **NOT** implicitly require the expected type.  Use the `type` keyword to explicitly constrain the type.
+
+Note that the `type` keyword allows `"integer"` as a value for convenience, but keyword and format applicability does not recognize integers as being distinct from other numbers because [[RFC7159|JSON]] itself does not make that distinction.  JSON Schema defines integers [mathematically](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-6.3), meaning that both `1` and `1.0` are considered to be integers for the purpose of the `type` keyword.
```

Comment:
```suggestion
Note that the `type` keyword allows `"integer"` as a value for convenience, but JSON Schema keyword and format applicability does not recognize integers as being distinct from other numbers because [[RFC7159|JSON]] itself does not make that distinction.  JSON Schema defines integers [mathematically](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-6.1.1) as "any number with a zero fractional part", meaning that both `1` and `1.0` are considered to be integers for the purpose of the `type` keyword.
```

Reply:
Section 6.3 "Mathematical Integers" is correct here.  There is no 6.1.1 in that draft of JSON Schema.

I am against "any number with a zero fractional part" as a description as it is ambiguous as to whether it refers to the value or the textual representation.

Again, this is a JSON Schema thing.  JSON Schema's wording is what matters, and OpenAPI ***CANNOT*** change the requirement.  The only thing we can or should do here is reference the appropriate section of JSON Schema.

Reply:
Quite right that there is no 6.1.1 in that draft. Let me see if I can find what I was looking at. In any event, I don't believe section 6.3 "defines integers -- this is the problem I was trying to fix, so let's find a good solution to that.

Reply:
Here's the correct link:

https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#autoid-11

Reply:
JSON Schema _Validation_ defines `"integer"` in [section 6.1.1](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-6.1.1)

Reply:
I've updated my suggested change. @handrews please review.

Reply:
@mikekistler the point of this link is to explain why `1.0` is an integer.  The core spec §6.3 does that.  That is literally the entire purpose of that section.

This text here already explains that the `type` keyword allows `"integer"`.  If you _also_ want to link to the `type` keyword's section, we can do that on the word `type` (although I'd rather not- we don't do that for any other mention of any other JSON Schema keyword AFAICT so it would be a confusing inconsistency).

If you want to look up the details of `type`, it's obvious that you should look up its section in the validation spec.  On the other hand, the mathematical nature of integers in JSON Schema is a recurring point of confusion, and it's less obvious where to resolve it.  Hence linking to the section that explains it.

Reply:
@handrews When I follow the link to section 6.3 I see this text:
> Some programming languages and parsers use different internal representations for floating point numbers than they do for integers.
> For consistency, integer JSON numbers SHOULD NOT be encoded with a fractional part.

I don't think there is anything here that "defines integers", and I don't know why I would conclude from this text that "1.0" is an integer.

Reply:
@mikekistler OK let me think on this more.  §6.1.1 of the validation spec still isn't what should go here.